### PR TITLE
ensure set_embedding_id_to_latest sets the newest

### PR DESCRIPTION
**Pull request by japrescott**
closes lig-1971
- ensure `set_embedding_id_to_latest` actually sets to the latest

### CODE REVIEW

File: lightly/api/api_workflow_upload_embeddings.py [hunk 1]
```diff
@@ -41,7 +41,10 @@ def set_embedding_id_to_latest(self):
             self._embeddings_api.get_embeddings_by_dataset_id(
                 dataset_id=self.dataset_id
             )
-        self.embedding_id = embeddings_on_server[-1].id
+        if len(embeddings_on_server) == 0:
+            raise RuntimeError(f"There are no known embeddings for dataset_id {self.dataset_id}.")
+        # return first entry as the API returns newest first
+        self.embedding_id = embeddings_on_server[0].id
```
Comment (IgorSusmelj) [inline lightly/api/api_workflow_upload_embeddings.py:47]: The entries we get from the API are sorted by what? (Not timestamp and latest entry is last?)

**Review by IgorSusmelj — APPROVED** [review]
LGTM. I assume you doubled checked what the API returns and in which order?

File: lightly/api/api_workflow_upload_embeddings.py (hunk 1 above)
Comment (japrescott) [reply lightly/api/api_workflow_upload_embeddings.py:47]: `.const embeddings = await db.collection<IMongoEmbeddingSummary>('embeddingSummaries').find({ datasetId: toMongoId(datasetId) }).sort({ createdAt: -1 }).toArray();` thats how its called in the DB.

Don't know how to test it properly. Maybe you could upload multiple embeddings and verify that it always returns the first created embedding? Then we know this works as expected

Comment (MalteEbner) [reply lightly/api/api_workflow_upload_embeddings.py:47]: I would unittest this in the typescript tests an ensure it there.

**Review by MalteEbner — APPROVED** [review]
The change looks good to me, great that this error case now has a proper error message.

File: lightly/api/api_workflow_upload_embeddings.py (hunk 1 above)
Comment (MalteEbner) [reply lightly/api/api_workflow_upload_embeddings.py:47]: Edit: I would unittest the order in typescript.

Comment (japrescott) [reply lightly/api/api_workflow_upload_embeddings.py:47]: verified

**Pull request merged**